CMakeLists.txt

```diff
@@ -432,9 +432,11 @@ add_subdirectory(lib)
 add_subdirectory(third-party)
 add_subdirectory(src)
 add_subdirectory(examples)
-add_subdirectory(tests)
-#add_subdirectory(tests/testdata)
-add_subdirectory(integration-tests)
+if(BUILD_STATIC_LIBS)
```

Comment:
Ideally the control flow is a little better than this, since building static libs shouldn't imply that one wants to also build testing.

Can we make this either:
```
if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME AND BUILD_STATIC_LIBS)
  add_subdirectory(tests)
  #add_subdirectory(tests/testdata)
  add_subdirectory(integration-tests)
endif()
```

Or create a new CMake option, `BUILD_TESTING`, and do this:
```
if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME AND BUILD_TESTING)
  add_subdirectory(tests)
  #add_subdirectory(tests/testdata)
  add_subdirectory(integration-tests)
endif()
```

where `BUILD_TESTING` is defined as:
```
cmake_dependent_option(BUILD_TESTING "Enable tests" ON "BUILD_STATIC_LIBS" OFF)
```

I'm fine with either way. cc @tatsuhiro-t